src/services/cgnOperatorSearchService.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * This service interacts with the GCN operator search API
+ */
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ProblemJson,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Merchant } from "../../generated/cgn-operator-search/Merchant";
+import {
+  ResponseErrorStatusNotDefinedInSpec,
+  withCatchAsInternalError,
+  withValidatedOrInternalError
+} from "../utils/responses";
+import { readableProblem } from "../utils/errorsFormatter";
+import { CgnOperatorSearchAPIClient } from "../../src/clients/cgn-operator-search";
+import { OnlineMerchantSearchRequest } from "../../generated/io-cgn-operator-search-api/OnlineMerchantSearchRequest";
+import { OnlineMerchants } from "../../generated/cgn-operator-search/OnlineMerchants";
+import { OfflineMerchantSearchRequest } from "../../generated/io-cgn-operator-search-api/OfflineMerchantSearchRequest";
+import { OfflineMerchants } from "../../generated/cgn-operator-search/OfflineMerchants";
+import { IResponseType } from "italia-ts-commons/lib/requests";
+export default class CgnService {
+  constructor(
+    private readonly cgnOperatorSearchApiClient: ReturnType<
+      CgnOperatorSearchAPIClient
+    >
+  ) {}
+
+  private readonly toResponse = <T>(
+    response:
+      | IResponseType<200, T>
+      | IResponseType<404, undefined>
+      | IResponseType<500, ProblemJson>
+  ):
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<T> => {
+    switch (response.status) {
+      case 200:
+        return ResponseSuccessJson(response.value);
+      case 404:
+        return ResponseErrorNotFound("Not Found", "Operator Not found");
+      case 500:
+        return ResponseErrorInternal(readableProblem(response.value));
+      default:
+        return ResponseErrorStatusNotDefinedInSpec(response);
+    }
+  };
+
+  /**
+   * Get the CGN operator/merchant by its identifier.
+   */
+  public readonly getMerchant = (
```

Comment:
Maybe we have to protect this API with a check on the logged user CGN's activation. Only users with a valid CGN can see Merchant detail (including dicount codes)

Reply:
Done in https://github.com/pagopa/io-backend/pull/804/commits/8db01fa6d5fb9496230544bbfc6cbbb9e480229a